### PR TITLE
Filter unlibrary

### LINKED ISSUE #115: Filter unlibrary books
add the ability to filter books in unlibrary

### PR DESCRIPTION
**Pull request by drojasca**
# Summary
- Added ability to filter books in Unlibrary
- Generalized FilterMap to change what is displayed in the filter menu
- **Changed the minSdkVersion to 24**

# Screenshots
image

Closes #115

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -9,7 +9,7 @@ android {
 
     defaultConfig {
         applicationId "com.example.unlibrary"
-        minSdkVersion 21
+        minSdkVersion 24
```
Comment (calebschoepp) [inline app/build.gradle:12]: Not a problem, but just curious why you bumped this?

File: app/src/main/java/com/example/unlibrary/util/FilterMap.java [hunk 2]
```diff
@@ -80,11 +79,9 @@ public void setMap(Map<Book.Status, Boolean> map) {
...
-        return new boolean[]{
-                mFilter.get(Book.Status.AVAILABLE),
-                mFilter.get(Book.Status.REQUESTED),
-                mFilter.get(Book.Status.ACCEPTED),
-                mFilter.get(Book.Status.BORROWED),
-        };
+        Boolean[] filter = Arrays.stream(mFilter.keySet().toArray()).map(status -> mFilter.get(status)).toArray(Boolean[]::new);
+        boolean[] bfilter = new boolean[filter.length];
+        IntStream.range(0, bfilter.length).forEach(index -> bfilter[index] = filter[index].booleanValue());
+        return bfilter;
```
Comment (calebschoepp) [inline app/src/main/java/com/example/unlibrary/util/FilterMap.java:85]: Nice work, it's great to not have this hardcoded anymore

File: app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryRepository.java [hunk 3]
```diff
@@ -163,6 +204,7 @@ public void updateBook(Book book, OnSuccessListener<? super Void> onSuccessListe
                 .addOnFailureListener(onFailureListener);
     }
 
+
```
Comment (calebschoepp) [inline app/src/main/java/com/example/unlibrary/unlibrary/UnlibraryRepository.java:207]: Nit: Extra newline

**Review by calebschoepp — DISMISSED** [review]
Few comments but otherwise LGTM!

File: app/build.gradle (hunk 1 above)
Comment (drojasca) [reply app/build.gradle:12]: I needed to bump up the sdk version so I could use Array.stream(). The older version didn't have them.

**Review by calebschoepp — APPROVED** [review]
LGTM!

**Pull request merged**